catalog/task/create-pyxis-image/0.1/create-pyxis-image.yaml

```diff
@@ -0,0 +1,81 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: create-pyxis-image
+  labels:
+    app.kubernetes.io/version: "0.1"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: release
+spec:
+  description: >-
+    Tekton task that executes a python script to push a container image's metadata to Pyxis
+  params:
+    - name: pyxisUrl
+      type: string
+      description: The URL to the Pyxis API to be used, ending in `/`
+    - name: pyxisSecret
+      type: string
+      description: The kubernetes secret to use to authenticate to Pyxis
+    - name: certified
+      type: string
+      description: If set to true, the image will be marked as certified in its Pyxis entry
+      default: "false"
+    - name: tag
+      type: string
+      description: The tag to use when pushing the container image metadata to Pyxis
+    - name: isLatest
+      type: string
+      description: If set to true, the image will have a latest tag added with its Pyxis entry
+      default: "false"
+    - name: skopeoInspectFile
+      type: string
+      description: The filename of the saved skopeo inspect output stored on the input workspace
+      default: "skopeo-inspect.json"
+  results:
+    - name: containerImageID
+      description: ID of the created entry in Pyxis
+  workspaces:
+    - name: input
+      description: The workspace where the skopeoInspectFile is located
+  steps:
+    - name: create-pyxis-image
+      image:
+        quay.io/hacbs-release/release-utils@sha256:1cb213e052adc0919f0ef1eb7f0fbb0f1a3b87cb98024095d1e1d036e8f5c5ac
+      env:
+        - name: pyxisCert
+          valueFrom:
+            secretKeyRef:
+              name: $(params.pyxisSecret)
+              key: cert
+        - name: pyxisKey
+          valueFrom:
+            secretKeyRef:
+              name: $(params.pyxisSecret)
+              key: key
+      script: |
+        #!/usr/bin/env sh
+
+        if [[ -z "$(params.pyxisUrl)" || -z "$(params.pyxisSecret)" || -z "$(params.certified)" || \
+              -z "$(params.tag)" || -z "$(params.isLatest)" ]]
+        then
+          echo "One or more parameter(s) not provided"
+          exit 1
+        fi
+
+        echo "${pyxisCert}" > /tmp/crt
+        echo "${pyxisKey}" > /tmp/key
+
+        set -x
+
+        cd /home/release-utils/pyxis
+        PYXIS_CERT_PATH=/tmp/crt PYXIS_KEY_PATH=/tmp/key python3 create_container_image.py \
```

Comment:
If that script is in an image, create a binary without extension and execute it as such. If in the future we move from python3 to python4 or from python to go, we would have to modify the task as well.

Reply:
https://github.com/hacbs-release/release-utils/pull/14